### ISSUE TITLE
compile_predict leaves dropout and batch norm live inside a Scan, so recurrent inference is stochastic

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor-ml#124]
`rewrite_for_prediction` runs an `EquilibriumDB` of node rewriters that match outer `DropoutLayer` and `BatchNormLayer` nodes. A `Scan`'s inner graph is never rewritten, so for any recurrent model the one API whose whole contract is deterministic, batch-independent inference delivers neither. The noise is not subtle: on a three-layer recurrent cell two identical calls differ by nearly six times the mean output magnitude.

```python
import numpy as np
import pytensor.tensor as pt
from pytensor.scan import scan

from pytensor_ml.layers import BatchNorm, Dropout, Linear
from pytensor_ml.pytensorf import compile_predict

xseq = pt.tensor("xseq", shape=(None, None, 4))          # (time, batch, features)
lin, bn, dro = Linear("l", 4, 4), BatchNorm("bn", 4), Dropout("d", p=0.5, random_state=0)
out = scan(lambda xt: dro(bn(lin(xt))), sequences=[xseq], return_updates=False)

predict = compile_predict(out, inputs=[xseq])
X = np.random.default_rng(0).normal(size=(6, 8, 4))

print(np.allclose(predict(X), predict(X)))     # False -- two identical calls disagree
print(np.abs(predict(X) - predict(X)).max(), np.abs(predict(X)).mean())
# roughly 4.5 against 0.8: the spread between two calls is several times the signal
```

Descending into `Scan.inner_outputs` when rewriting would fix this.
